### ISSUE TITLE
Be more resilient to "multiple nodes exists" errors

### ISSUE THREAD

**Issue by ParetoOptimalDev** [mclear-tools/consult-notes#19]
Right now if I have multiple nodes and I type that nodes name this error pops up in messages:

```
Error in post-command-hook (vertico--exhibit): (user-error "Multiple nodes exist with title or alias \"compare size of haskell docker container and nix built haskell docker container\"")
```

Sometimes when this happens my the consult-buffer/vertico window becomes unresponsive in the way of nothing I type shows up and I can't seem to select anything. I have to `C-g` to get out of it and re-enter it. This doesn't always happen though.

I'm not sure if this is something to be fixed consult-notes side, vertico side, or perhaps something problematic in my config.

For some context, I ideally I wouldn't have duplicates but they occur sometimes because of:

- syncthing conflicts
-  making the wrong kind of links

New users of consult-notes with large messy org-roams might not want to spend a couple hours removing 50+ duplicates before using it either like I did :sweat_smile:

**Comment by mclearc** [comment]
> Right now if I have multiple nodes and I type that nodes name this error pops up in messages:
> 
> ```
> Error in post-command-hook (vertico--exhibit): (user-error "Multiple nodes exist with title or alias \"compare size of haskell docker container and nix built haskell docker container\"")
> ```

Is there some kind of error handling in org-roam that has been changed for this? 

> For some context, I ideally I wouldn't have duplicates but they occur sometimes because of:
> 
>     * syncthing conflicts
> 
>     * making the wrong kind of links
> 
> 
> New users of consult-notes with large messy org-roams might not want to spend a couple hours removing 50+ duplicates before using it either like I did 😅

Yes that's fair! I actually wasn't aware that org-roam allows for conflicts like this (I don't use org-roam anymore). consult-notes-org-roam largely uses org-roam functions to assemble all the info -- do you see similar errors when using completing read with org-roam without consult-notes?

**Comment by mclearc** [comment]
Do you have a way of reliably reproducing this error? I can't seem to generate it simply with duplicate files. Incidentally, when using duplicate files I get the error 

```
Warning (org-roam): SQL constraint violation: "UNIQUE constraint failed: nodes.id", 19, nil, "constraint failed" for NOTE(20210718T000844.283179) in /Users/roambot/org-roam/NOTE copy.org Disable showing Disable logging
```
So the database doesn't even get updated. Does this not happen to you?

**Comment by ParetoOptimalDev** [comment]
> Do you have a way of reliably reproducing this error? I can't seem to generate it simply with duplicate files. Incidentally, when using duplicate files I get the error

Try creating a node as "emacs" as a file. Then create a daily for today. Create a heading linking to emacs node. Now call org-attach with like C-c a f or call the command to generate a uuid.

Then call sync.

If I have time later I'll try and do this from emacs -Q.

**Comment by mclearc** [comment]
I'm going to close this for now. I'm not sure how to make these kinds of errors less intrusive and have had trouble reproducing the problem in a way that isn't just a problem with org-roam itself. But feel free to reopen if you have further suggestions, etc.

**Comment by ParetoOptimalDev** [comment]
No problem. I'm still affected by it every so often and might have some time to make a minimal repro before years end.

**Comment by leinfink** [comment]
Just wanted to add that this is still an issue, and I don't think it's a problem with org-roam itself. Org-roam nowadays seems to handle seeming "duplicates" nicely: One can have multiple nodes with the same title (the unique identifier is the org-ID, after all). I sometimes even do this intentionally.
`org-roam-node-find` works fine, but `consult-notes` gives the error mentioned above.

**Comment by pastnontra** [comment]
Yes, the problem is fixed. (I'm still very confused because I don't have any empty node in my testing environment though.) I can run the command and open the node without any error both in testing environment and my actual Org roam directory which contains lots of duplicate nodes.
I'll keep test if there is still issues and give feedback recently. But I don't use too much advanced Org roam functions such as alias, so the test scope is limited.

**Comment by pastnontra** [comment]
After trying to use `consult-notes` for a day, everything works fine. There still lots of `consult-notes` and Org roam features I haven't used, plus other thoughts about Org roam UX (I may fill another issue to talk about this), but as far as I know, none of them is directly related to this issue. So I think this issue can be closed for now.

However, I still have one last question about ID suffix
> This preserves the current UX for most users while making duplicate-title scenarios both functional and visually distinguishable. It may run into issues for folks with a very large number of files/nodes.

If I understand correctly, the ID suffix is not only a visual indicator, but also part of Title to ensure node is unique. 
In that cases, there seems no simple way to:
- give ID suffix a different (dimmer) face
- let query doesn't match ID suffix

at least in my limited knowledge of Elisp. It's a bit annoying, but I'm satisfy with current solution after all. 


Thanks for your patient. It's an amazing package.

**Comment by mclearc** [comment]
Closing this as resolved. The core crash was fixed in 26297ff (ID disambiguation for duplicate titles) and 037231a (empty title fallbacks). The remaining UX concern about ID suffixes being searchable is a minor trade-off of the disambiguation approach -- the ID is part of the candidate string so completion frameworks will match against it. Thanks to @pastnontra for the detailed repro and testing.
